userspace/libsinsp/sinsp_filtercheck_fspath.cpp

```diff
@@ -240,6 +240,56 @@ std::unique_ptr<sinsp_filter_check> sinsp_filter_check_fspath::allocate_new()
 	return ret;
 }
 
+std::string sinsp_filter_check_fspath::parse_dirfd_stateless(sinsp_evt *evt, std::string_view name, int64_t dirfd)
```

Comment:
If i understand correctly, all we need is to skip the `evt->set_fd_info(evt->get_tinfo()->get_fd(dirfd));` call in `parse_dirfd`, right? 
I think we can just add a boolean optional parameter (that of course defaults to `true` to `parse_dirfd`, like:
```
std::string sinsp_parser::parse_dirfd(sinsp_evt *evt, std::string_view name, int64_t dirfd, bool update_tinfo=true)
```
and pass `false` when called from the filterchecks.

Reply:
Correct, can do that. Are we sure we want one more if statement in the parsers logic?